From 250ab66f58aa37193f2e8b7814946e7a43d25cc7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ondrej=20Mosn=C3=A1=C4=8Dek?= <omosnacek@gmail.com>
Date: Fri, 17 Nov 2023 10:48:40 +0100
Subject: [PATCH] src/CMakeLists.txt - set VERSION and SOVERSION on the library
 target
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The version variables are currently only passed to ecm_setup_version(),
but they need to be also specified as target properties so that CMake
installs versioned .so files on Linux, which is required for proper
distribution packaging.

Signed-off-by: Ondrej Mosnáček <omosnacek@gmail.com>
---
 src/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d588524..4b983c8 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -23,7 +23,10 @@ add_library(
     KDAB::kdsingleapplication ALIAS kdsingleapplication
 )
 set_target_properties(
-    kdsingleapplication PROPERTIES OUTPUT_NAME "kdsingleapplication${KDSingleApplication_LIBRARY_QTID}"
+    kdsingleapplication
+    PROPERTIES VERSION ${${PROJECT_NAME}_VERSION}
+               SOVERSION ${${PROJECT_NAME}_SOVERSION}
+               OUTPUT_NAME "kdsingleapplication${${PROJECT_NAME}_LIBRARY_QTID}"
 )
 
 target_include_directories(
